Add ADR about TAD standard (#107)
MinBZK/ai-validation: docs/Projects/TAD/ADRs/0002-tad-reporting-standard.md

> # TAD-0002 TAD Reporting Standard

Reviewer: I would add the TAD ADR in the TAD repo. Else we get a mess in all the ADRs and where they are.

Reviewer 2: We don't really have a "TAD repo", or do we?

Reviewer: We do not but i would not add all ADRs in this repo. Would prefer to keep them as close as possible to the project.

Author: Decided to keep the current structure for now.

## Context

The TAD Reporting Standard proposes a standardised way of capturing information of ML-models and systems.

## Assumptions

> There does not exists a standard of capturing all relevant information on ML-models that also includes

Reviewer 3: ```suggestion
There is no exists a standard of capturing all relevant information on ML-models that also includes
```

fairness and bias tests and regulatory assessments.

A widely used implementation for [Model Cards for Model Reporting](https://arxiv.org/abs/1810.03993) is given
by the [Hugging Face Model Card metadata specification](https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1),
which in turn is based on [Papers with Code Model Index](https://github.com/paperswithcode/model-index).

> This implementation does not include fields for bias and fairness tests or regulartory assessments.

Reviewer 3: ```suggestion
This implementation does not include fields for measurements and metrics from technical tests like bias and fairness tests or regulatory assessments.
```

Author: It does include fields for metrics, only less elaborate than our metric field. How about: 
```suggestion
This implementation does not capture sufficient details about metrics and does not include measurements from technical tests on bias en fairness or regulatory assessments.
```
?

Reviewer 3: Yes, very good, I can not edit your suggestion but it should say: 'bias and fairness'.. going Duch there ;)

## Decision

We decided to implement a custom reporting standard. Our reporting standard can be split up
into three elements.

1. System Card, containing information about a group of ML-models which accomplish a specific task.
A System Card can refer to multiple Model Cards, either a Model Card specified by the TAD Reporting
Standard, or any other model card. A System Card can refer to multiple Assessment Cards.

> 2. Model Card, containing information about a specific ML-model.

Reviewer 3: I wonder if we also make measurements and metrics separate includes, so results in their own file, as we also do with assessments.

Reviewer 2: We may want to, in general, accept includes, and say something like that here too.

This is what we currently say [in our standard](https://minbzk.github.io/ai-validation/Projects/TAD/Reporting-Standard/#introduction):

> These model_cards and assessment_cards can be included verbatim into a system_card, or referenced with an !include statement, allowing for minimal cards to be compact in a single file. Extensive cards can be split up for readability and maintainability. Our standard allows for the !include to be used anywhere.

Author: In principle you can !include inside the system card any field you want. The standard prefers to do this for in any case the model card and assessment card. If we also want to prefer to include metrics and measurements via an include mechanism, we should specify this in the TAD standard. So for this ADR I would not make these suggestions.

Reviewer 3: Ok, awesome, should have seen that ;)

3. Assessment Card, containing information about a regulatory assessment.

We were heavily inspired by the Hugging Face Model Card metadata specification, which we essentially
extended to allow for:

1. More finegrained information on performance metrics.
2. Capturing additional measurements on fairness and bias.
3. Capturing regulartory assessments.

The extension is not strict, meaning that there the TAD Reporting Standard is not a valid Hugging Face

> metadata specification. The reason for this is that some fields in the Hugging Face standard to too much

Reviewer 3: ```suggestion
metadata specification. The reason for this is that some fields in the Hugging Face standard are too much
```

intertwined with the Hugging Face ecosystem and it would not be logical for us to couple our implementation
this thighly to Hugging Face.

> The standard is written in yaml.

Reviewer 2: I'm not sure we need to say this. The standard is more general, we just happen to have an implementation in yaml
